thamos/lib.py

```diff
@@ -459,16 +463,25 @@ def get_analysis_results(api_client: ApiClient, analysis_id: str):
     if analysis_id.startswith("package-extract-"):
         api_instance = ImageAnalysisApi(api_client)
         method = api_instance.get_analyze
+        response = _retrieve_analysis_results(method, analysis_id)
+        return response.result
     elif analysis_id.startswith("provenance-checker-"):
         api_instance = ProvenanceApi(api_client)
         method = api_instance.get_provenance_python
+        response = _retrieve_analysis_result(method, analysis_id)
+        return response.result["report"], response.result["error"]
     elif analysis_id.startswith("adviser-"):
         api_instance = AdviseApi(api_client)
         method = api_instance.get_advise_python
+        response = _retrieve_analysis_result(
+            api_instance.get_advise_python, analysis_id
+        )
+        return response.result, response.result["error"]
```

Comment:
Why did you put this cone snipped to each branch? The same code gets called after each branch. Also it might be good idea to be consistent with the method signature - what gets returned.

Reply:
This was to be consistent with the return values of advise, provenance check, and image analysis

Reply:
Each ones' return values are slightly different

Reply:
Oh, I missed the image analysis part. OK in that case :+1: